src/Test/StateMachine/Markov.hs

```diff
@@ -285,24 +288,54 @@ toMatrix m = do
 
 ------------------------------------------------------------------------
 
+instance IsString str => MonadFail (Either str) where
+  fail = Left . fromString
+
+handle :: Maybe a -> String -> Either String a
```

Comment:
This combinator is called `maybeToRight` in https://hackage.haskell.org/package/either-5.0.1/docs/Data-Either-Combinators.html